promscale/troubleshooting.md

```diff
@@ -0,0 +1,92 @@
+# Troubleshooting Promscale
+This section contains some ideas for troubleshooting common problems experienced
+with Promscale.
+
+<!---
+* Keep this section in alphabetical order
+* Use this format for writing troubleshooting sections:
+ - Cause: What causes the problem?
+ - Consequence: What does the user see when they hit this problem?
+ - Fix/Workaround: What can the user do to fix or work around the problem? Provide a "Resolving" Procedure if required.
+ - Result: When the user applies the fix, what is the result when the same action is applied?
+* Copy this comment at the top of every troubleshooting page
+-->
+
+## Backing up Promscale data gives an `out of shared memory` error
+Promscale creates a table for each metric. When you run `pg_dump` and
+`pg_restore`, they lock each table to work on it. This can require a lot of
+locks in a single transaction, which can quickly grow higher than the maximum
+number of locks allowed. If the number of locks required exceeds the number
+allowed, Promscale gives an `out of shared memory` error, and the action is
+aborted.
+
+To fix this error, increase the `max_locks_per_transaction` setting in the
+PostgreSQL configuration file, so that `pg_dump` and `pg_restore` can create the
+required number of locks, and the backup or restore can continue.
+
+For more information, see the
+[backup and restore documentation][promscale-backup-restore].
+
+## Data is occupying too much space
+Promscale keeps metric data in chunks. The most recent chunk is kept
+uncompressed as a cache for faster querying, and chunks are compressed as they
+age. If your data is taking up more disk space than expected, compression might
+not be working correctly. If compression is not working correctly, older chunks
+might not be getting compressed, and they might be taking up too much room on disk. To check
+that compression is working correctly, query the `prom_info.metric` view and
+make sure that `total_chunks-number_compressed_chunks` is not larger than 2.
+
+If compression is not working correctly, make sure that you have enough
+background workers to do scheduled jobs like compression and retention. To
+calculate the number of background workers you require, start with the number of
+databases you have in your environment, plus 2. If you are using TimescaleDB
+version 2.0.0 or earlier, make sure that you are running the maintenance `cron`
+jobs, and that they are returning success. When compression is working
+correctly, your data is being compressed as it ages.
```

Comment:
@ramonguiu should we add configuring the default chunk_interval from `8h` to `2h` or `4h` to reduce the space consumption as it compresses the uncompressed data quicker than the defaults.

Reply:
I don't think I would recommend changing the chunk_interval for now since this has other consequences.